### PR TITLE
Add per-operation locks to FileManager

### PR DESCRIPTION
**Pull request by matrss**
Without these locks there is a race condition when writing to the operations "main.ftml" file, which happens in multiple different methods, e.g. save_file.  When the mscolab application is running in a multi-threaded wsgi server it can happen that two threads try to write to the same file, corrupting it in the process.

Split off from #2100.

### CODE REVIEW

File: mslib/mscolab/file_manager.py [hunk 1]
```diff
@@ -29,6 +29,7 @@
 import difflib
 import logging
 import git
+import threading
```
Comment (ReimarBauer) [inline mslib/mscolab/file_manager.py:32]: please import only what we need from threading 
  
 from threading import Lock

Comment (matrss) [reply mslib/mscolab/file_manager.py:32]: Why? I prefer having it namespace'd properly where I use it and it is not like `threading` is excessively long. Having it just be called `Lock` where it is used makes it ambiguous if it is coming from threading or from multiprocessing for example.

Comment (ReimarBauer) [reply mslib/mscolab/file_manager.py:32]: good point, we keep it. 
We have no generel style for this, usually we use import when we use lots of its methods.

in both cases the module is always fully imported (no speedup by a selection, python parses the whole module).

**Review by ReimarBauer — APPROVED** [review]
Thx

**Pull request merged**